### PR TITLE
Update Accelerator Connector for Registry

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Update Accelerator Connector for Registry

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 1]
```diff
@@ -492,7 +497,12 @@ def select_cluster_environment(self) -> ClusterEnvironment:
 
     def set_distributed_mode(self, distributed_backend: Optional[str] = None):
 
-        if distributed_backend is not None:
+        if distributed_backend is None and self.is_training_type_in_plugins:
+            return
+
+        if distributed_backend is not None and distributed_backend in TrainingTypePluginsRegistry:
+            self.distributed_backend = TrainingTypePluginsRegistry[distributed_backend]["distributed_backend"]
```
Comment (SeanNaren) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:504]: Could you explain what's happening here @kaushikb11?

Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:504]: Yup, when you go through the `set_distributed_mode` function, you will see that there are certain decisions being made based on `distributed_backend`. Hence, I am assigning `distributed_backend` from the Registry. @SeanNaren

Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:504]: The flow goes like this for plugin `ddp_find_unused_parameters_false`:
handle_plugins(`ddp_find_unused_parameters_false`) => set_distributed_mode(`ddp_find_unused_parameters_false`) => set distributed_backend(`ddp`) => decisions based on `ddp`.

Comment (SeanNaren) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:504]: Any reason we can't do `self.distributed_backend = TrainingTypePluginsRegistry[distributed_backend]` this should assign a plugin right?

Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:504]: @SeanNaren 

```python
from pytorch_lightning.plugins import TrainingTypePluginsRegistry

print(TrainingTypePluginsRegistry["ddp_find_unused_parameters_false"])
```

{'description': 'DDP Plugin with `find_unused_parameters` as False', 'init_params': {'find_unused_parameters': False}, 'plugin': <class 'pytorch_lightning.plugins.training_type.ddp.DDPPlugin'>, 'distributed_backend': 'ddp'}

Comment (SeanNaren) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:504]: Oh of course my bad :)

Given the naming and hopefully eventually cleaning up such that `distributed_backend` becomes clearer in it's responsibility, this is fine!

**Review by SeanNaren — APPROVED** [review]

**Review by tchaton — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/trainer/connectors/accelerator_connector.py [hunk 2]
```diff
@@ -310,6 +311,10 @@ def parallel_devices(self) -> List[Union[torch.device, int]]:
     def root_gpu(self) -> Optional[int]:
         return self.accelerator.root_device.index if not isinstance(self.accelerator, TPUAccelerator) else None
 
+    @property
+    def is_training_type_in_plugins(self) -> bool:
+        return any(isinstance(plug, str) and plug in TrainingTypePluginsRegistry for plug in self.plugins)
+
```
Comment (ananthsub) [inline pytorch_lightning/trainer/connectors/accelerator_connector.py:317]: n00b question: could you explain why this iterates over all plugins in the connector? from the naming i'd assume this meant to check if a single plugin instance is contained in the registry

Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:317]: Yup

Comment (kaushikb11) [reply pytorch_lightning/trainer/connectors/accelerator_connector.py:317]: Yup!

**Pull request merged**